src/System.Private.CoreLib/shared/System/ComponentModel/DefaultValueAttribute.cs

```diff
@@ -48,9 +52,39 @@ public DefaultValueAttribute(Type type, string value)
                 {
                     _value = Convert.ChangeType(value, type, CultureInfo.InvariantCulture);
                 }
+
+                return;
+
+                // Looking for object TypeDescriptor.InternalConvertFromInvariantString(Type, string) and call to conversion function ConvertFromInvariantString
+                bool TryConvertFromInvariantString(Type typeToConvert, string stringValue, out object conversionResult)
+                {
+                    conversionResult = null;
+
+                    // lazy init reflection objects
+                    if (s_convertFromInvariantString == null)
+                    {
+                        Type typeDescriptorType = Type.GetType("System.ComponentModel.TypeDescriptor, System, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089", throwOnError: false);
+
+                        Delegate convertFromInvariantString = null;
+                        if (typeDescriptorType != null)
+                        {
+                            convertFromInvariantString = Delegate.CreateDelegate(typeof(Func<Type, string, object>), typeDescriptorType, "InternalConvertFromInvariantString", ignoreCase: false, throwOnBindFailure: false);
+                        }
```

Comment:
@jkotas @ericstj  added this check to be sure that helper method exists on "corelib users". I'm not sure if this makes sense, is corefx the only "user" of coreclr today and in future?

Reply:
`if (typeDescriptorType != null)` check is fine. It is the whole point of the fallback path.

For `Delegate.CreateDelegate`, I do not think we need `throwOnBindFailure: false`. If `System.ComponentModel.TypeDescriptor` exists, we do expect the helper method to exist too.

Reply:
> we should expect that the helper method exists too.

My fault i wasn't clear, but you answered! If i remove `throwOnBindFailure: false` i can collapse all on assignment...if `InternalConvertFromInvariantString` will miss throw exception.

```cs
// lazy init reflection objects
if (s_convertFromInvariantString == null)
{
   Type typeDescriptorType = Type.GetType("System.ComponentModel.TypeDescriptor, System, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089", throwOnError: false);
   Volatile.Write(ref s_convertFromInvariantString, typeDescriptorType == null ? new object() : Delegate.CreateDelegate(typeof(Func<Type, string, object>), typeDescriptorType, "InternalConvertFromInvariantString", ignoreCase: false));
}
```

Reply:
done